scipy/optimize/_root_scalar.py

```diff
@@ -0,0 +1,236 @@
+"""
+Unified interfaces to root finding algorithms for real or complex
+scalar functions.
+
+Functions
+---------
+- root : find a root of a scalar function.
+"""
+from __future__ import division, print_function, absolute_import
+
+from scipy._lib.six import callable
+
+from . import zeros as optzeros
+
+__all__ = ['root_scalar']
+
+
+class MemoizeDer(object):
+    """ Decorator that caches the value and derivative(s) of function each
+    time it is called. """
```

Comment:
Would be useful to clarify here that the size of the memoized set is 1; i.e. this only does something when calling the function with the same `x` 2x in a row. And that it's assumed that `args` does not change between function calls.